firefox-ios/Client/Frontend/Library/Bookmarks/Edit Bookmark/EditBookmarkViewController.swift

```diff
@@ -76,8 +76,19 @@ class EditBookmarkViewController: UIViewController,
     override func viewDidLoad() {
         super.viewDidLoad()
         title = .Bookmarks.Menu.EditBookmarkTitle
-        viewModel.onFolderStatusUpdate = { [weak self] in
-            self?.tableView.reloadSections(IndexSet(integer: Section.folder.rawValue), with: .automatic)
+        viewModel.onFolderStatusUpdate = { [weak self] shouldUpdateNewFolderSection in
+            guard let self = self else { return }
+
+            self.tableView.performBatchUpdates {
+                if shouldUpdateNewFolderSection {
+                    if self.viewModel.isFolderCollapsed {
+                        self.tableView.deleteSections(IndexSet(integer: Section.newFolder.rawValue), with: .none)
+                    } else {
+                        self.tableView.insertSections(IndexSet(integer: Section.newFolder.rawValue), with: .none)
+                    }
+                }
+                self.tableView.reloadSections(IndexSet(integer: Section.folder.rawValue), with: .automatic)
+            }
```

Comment:
haven't been too closely involved in the bookmarks realm. It seems we're using view models and also not using diffable datasource. Is there a specific reason we decided to opt out of following our current architecture pattern using Redux states, seems this is a newly created class.

Also curious what do you think about using diffable datasource which can automatically calculate the diff and update the table views accordingly instead of explicitly calling these table view methods?

Reply:
I made the decision to not include Redux for the Bookmarks refactor after hearing Orla say that we were not going to use it for existing features. 
Although this project is being managed like a new feature, it's UI is very close to what already exists for Bookmarks. If you go to the epic you can see the Figma for this project and get an idea for what I mean. 

You could make an argument that since we are duplicating views in some cases we could have used Redux anyway, but at the time I wasn't sure how much we were going to be duplicating new/legacy views versus doing inline checks in existing views, so I opted out of using Redux for this due to the added complexity and Orla's previous comments which I felt justified it. Happy to hop on a call to discuss more about this though!

I think diffable datasource would be a good candidate here, thanks for pointing that out! I will discuss with @MattLichtenstein about it.

Reply:
thanks for the context @dataports! sounds good to me. great! let me know when this PR is ready for another review.